### DISCUSSION TITLE
how to write loco deck one-wire memory?

### DISCUSSION THREAD

**Discussion by justinleeyang** [bitcraze/discussions#1669] [General]
Hi, 

How to write loco deck one-wire memory? In the figure below, is the value written correct? mems[0].pins = ?

![image](https://github.com/user-attachments/assets/9ad07630-7b79-42e6-a721-6317d06da7aa)
```
       
            mems[0].vid = 0xBC
            mems[0].pid = 0x06
            mems[0].name = 'bcLoco'

            board_name_id = OWElement.element_mapping[1]
            board_rev_id = OWElement.element_mapping[2]
 
            mems[0].elements[board_name_id] = 'loco deck'
            mems[0].elements[board_rev_id] = 'D'
            mems[0].pins = 0    
            mems[0].write_data(self._data_written)


```

**Comment by gemenerik** [comment]
The one-wire memory can be programmed using examples in our libraries. You can find the data to write in the deck drivers. Hope that helps.

**Reply by justinleeyang** [reply]
ok, how to define mems[0].pins ?
